bridge/pkg/terra/watcher.go

```
@@ -0,0 +1,224 @@
+				logger.Info("\ttoken: ", zap.Any("token", token))
+				logger.Info("\tsender: ", zap.Any("sender", sender))
+				logger.Info("\trecipient: ", zap.Any("recipient", recipient))
+				logger.Info("\tamount: ", zap.Any("amount", amount))
+				logger.Info("\tnonce: ", zap.Any("nonce", nonce))
+
+				senderAddress, err := StringToAddress(sender.String())
+				if err != nil {
+					logger.Error("cannod decode hex ", zap.Any("value", sender.String()))
+					continue
+				}
+				recipientAddress, err := StringToAddress(recipient.String())
+				if err != nil {
+					logger.Error("cannod decode hex ", zap.Any("value", recipient.String()))
+					continue
+				}
+				tokenAddress, err := StringToAddress(token.String())
+				if err != nil {
+					logger.Error("cannod decode hex ", zap.Any("value", token.String()))
+					continue
+				}
+				txHashValue, err := StringToHash(txHash.String())
+				if err != nil {
+					logger.Error("cannod decode hex ", zap.Any("value", txHash.String()))
+					continue
+				}
+				lock := &common.ChainLock{
+					TxHash:        txHashValue,
+					Timestamp:     time.Now(), // No timestamp available, consider adding it into transaction logs or request additionally from the blockchain
+					Nonce:         uint32(nonce.Uint()),
+					SourceAddress: senderAddress,
+					TargetAddress: recipientAddress,
+					SourceChain:   vaa.ChainIDTerra,
+					TargetChain:   vaa.ChainID(uint8(targetChain.Uint())),
+					TokenChain:    vaa.ChainID(uint8(tokenChain.Uint())),
+					TokenAddress:  tokenAddress,
+					TokenDecimals: uint8(tokenDecimals.Uint()),
+					Amount:        new(big.Int).SetUint64(amount.Uint()),
+				}
+				e.lockChan <- lock
+			}
+
+			// Query and report guardian set status
+			resp, err := http.Get(fmt.Sprintf("%s/wasm/contracts/%s/store?query_msg={\"guardian_set_info\":{}}", e.urlLCD, e.bridge))
+			if err != nil {
+				logger.Error("query guardian set error", zap.Any("error", err))
+				return
```

why not return the error here?